packages/react-native/Libraries/StyleSheet/processBackgroundImage.js

```diff
@@ -11,23 +11,24 @@
 'use strict';
 
 import type {ProcessedColorValue} from './processColor';
-import type {GradientValue} from './StyleSheetTypes';
-
+import type {BackgroundImageValue} from './StyleSheetTypes';
 const processColor = require('./processColor').default;
-const DIRECTION_KEYWORD_REGEX =
+
+// Linear Gradient
+const LINEAR_GRADIENT_DIRECTION_REGEX =
```

Comment:
Heads up, that we are starting soon on iOS to experiment with the native parser which replaced ViewConfig processors. I didn't port this for `backgroundImage` yet, but the other processors should all be there with tests. https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/react/renderer/css/CSSTransformOrigin.h

This is building in OSS now, but IDK the story for GTest. @cortinico would know better.

Reply:
@NickGerleman yes! i saw this one. i didn't spend much time to try running it as currently the flag was off. But i will try to write a native parser for gradients 😄. 

For now, i can split this PR into three (iOS, android and JS changes)